src/go/rpk/pkg/cli/cloud/cluster/select.go

```diff
@@ -71,5 +84,8 @@ Or:
 	}
 
 	cmd.Flags().StringVar(&profileName, "profile", "", fmt.Sprintf("Name of a profile to create or update (avoids updating %q)", profile.RpkCloudProfileName))
+	cmd.Flags().BoolVar(&serverlessPrivate, "serverless-private", false, "Use private networking for serverless clusters (if cluster supports it)")
+	cmd.Flags().BoolVar(&serverlessPublic, "serverless-public", false, "Use public networking for serverless clusters (if cluster supports it)")
+	cmd.MarkFlagsMutuallyExclusive("serverless-private", "serverless-public")
```

Comment:
what do you think of `--serverless-network` and then have `public` `private` as options and have a way to autocomplete the input? with RegisterFlagCompletionFunc like https://github.com/redpanda-data/redpanda/blob/1f135be36398fdb338a470f605cb740f13f7d501/src/go/rpk/pkg/cli/security/user/update.go#L71

This is just to avoid having 2 new flags and keep it tidy